lib/go-tc/tovalidate/rules.go

```diff
@@ -217,3 +217,14 @@ func StringIsValidFloat() *validation.StringRule {
 		return err == nil && !math.IsNaN(validated)
 	}, "must be a valid float")
 }
+
+// IsValidIPorCIDR returns whether the input is a valid IP address or CIDR notation subnet
+func IsValidIPorCIDR(input string) bool {
+	ip := net.ParseIP(input)
+	if ip != nil {
+		return true // Valid IP address
+	}
+
+	_, _, err := net.ParseCIDR(input)
+	return err == nil // Valid CIDR notation subnet
```

Comment:
Should we return the error from here?

Reply:
now removed as i removed post method itself.